src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts

```diff
@@ -585,7 +587,13 @@ class WordHighlighter {
 		// multi-doc ON
 		for (const editor of currentEditors) {
 			const tempModel = editor.getModel();
-			if (tempModel && tempModel !== model) {
+
+			const isValidModel =
```

Comment:
I think we can give all the open documents to the `MultiModelOccurenceRequest` and in `MultiModelOccurenceRequest` we filter the other models based on provider's selectors.

Reply:
+1  don't filter based on schemes like this but use the selector of the respective providers